usl_pipeline/cloud_functions/main.py

```diff
@@ -289,15 +290,31 @@ def _build_feature_matrix_from_archive(
                 continue
 
             name = pathlib.PurePosixPath(member.name).name
+            # TODO: Group logic branch by path (per hazard model)
             if name == file_names.ELEVATION_TIF:
                 return _read_elevation_features(fd)
+            # Handle heat model input files (WPS)
+            elif name.startswith("met_em") and name.endswith(".nc"):
+                return _read_wps_features(fd)
             # TODO: handle additional archive members.
             else:
                 logging.warning(f"Unexpected member name: {name}")
 
     return None, FeatureMetadata()
 
 
+def _read_wps_features(fd: IO[bytes]) -> Tuple[NDArray, FeatureMetadata]:
+    # Ignore type checker error - BytesIO inherits from expected type BufferedIOBase
```

Comment:
FYI you can use `typing.cast` to cast this as a BufferedIOBase , which would obviate the `# type: ignore` and also this comment :)
I don't have a strong preference between the two options, just pointing it out if you haven't seen it before.
https://docs.python.org/3/library/typing.html#typing.cast

Reply:
Tried both but ran into issues..

When I casted into BufferedIOBase, the test fails due to failure to load the dataset with xarray and changing the function signature to expect BufferedIOBase fails the type checker!